File: Assets/Disease Destroyer Assets/Scripts/pushLightScript.js
#pragma strict
var duration : float;
private var pushTime : float;

light.type = LightType.Spot;

function Start () {

}

function Update () {
	if(pushTime > 0){
		var frac = (0.5 - pushTime) * 2;
		//light.spotAngle = 108.0 * frac;
		light.range = 300.0 * frac;
		pushTime = pushTime - Time.deltaTime;
	}
	else{
		//light.spotAngle  = 0.0;
		light.range = 0.0;
	}
}

function push(){
	pushTime = duration;
}

File: Assets/Disease Destroyer Assets/Scripts/pushLightScript.cs
using UnityEngine;

public class pushLightScript : MonoBehaviour
{
    public float duration = .5f;
    float pushTime;
    Light source;
    void Awake() { source = GetComponent<Light>(); if (source) source.type = LightType.Spot; }
    void Update() { if (!source) return; if (pushTime > 0) { source.range = 300f * ((.5f - pushTime) * 2f); pushTime -= Time.deltaTime; } else source.range = 0; }
    public void push() => pushTime = duration;
}
